File: src/main/kotlin/net/rk4z/beacon/java/EventLogger.java
package net.rk4z.beacon.java;

import java.util.ArrayList;
import java.util.Collections;
import java.util.List;

public class EventLogger {
    private static final List<String> eventLog = Collections.synchronizedList(new ArrayList<>());

    public static void logEvent(Event event) {
        eventLog.add("Event: " + event.getClass().getSimpleName() + " at " + System.currentTimeMillis());
    }

    public static List<String> getEventLog() {
        synchronized (eventLog) {
            return new ArrayList<>(eventLog);
        }
    }
}

File: src/main/kotlin/net/rk4z/beacon/EventLogger.kt
package net.rk4z.beacon

/**
 * Singleton object that logs events.
 * It provides methods to log an event and get the event log.
 */
object EventLogger {
    // List to store the event log
    private val eventLog: MutableList<String> = mutableListOf()

    /**
     * Logs an event.
     * The event's class name and the current time in milliseconds are added to the event log.
     * @param event The event to log.
     */
    @JvmStatic
    fun logEvent(event: Event) {
        eventLog.add("Event: ${event::class.simpleName} at ${System.currentTimeMillis()}")
    }

    /**
     * Returns the event log.
     * @return A list of strings representing the event log.
     */
    @JvmStatic
    fun getEventLog(): List<String> {
        return eventLog.toList()
    }
}